api/lib/subscription/slashing.js

```diff
@@ -38,20 +53,58 @@ class SlashingMonitor {
     )
 
     // requires some more logic to not spam the notifications if a validator is down for 1000 blocks
-    // this.client.subscribe(
-    //   { query: 'liveness.missed_blocks >= 1' },
-    //   (response) => {
-    //     const missedBlocks = response.events['liveness.address'].map(
-    //       (address, index) => ({
-    //         operatorAddress: address,
-    //         missedBlocks: response.events['liveness.missed_blocks'][index],
-    //         height: response.height
-    //       })
-    //     )
-    //     console.log('Missed block', missedBlocks)
-    //   }
-    // )
+    this.client.subscribe(
+      { query: `tm.event='NewBlockHeader' AND liveness.missed_blocks >= 1` },
+      async (response) => {
+        try {
+          const missedBlocks = response.events['liveness.address'].map(
+            (address, index) => ({
+              networkId: this.networkId,
+              operatorAddress: address,
+              blocks: response.events['liveness.missed_blocks'][index],
+              reason: 'missed_blocks',
+              height: response.data.value.header.height,
+            })
+          )
+          const rows = await Promise.all(missedBlocks.map(async missedBlockEvent => {
+            const lastMissedBlockEvent = await getLastMissedBlock(this.networkId, missedBlockEvent.operatorAddress)
+            return aggregateMissedBlocks(lastMissedBlockEvent, missedBlockEvent)
+          }))
+          database(config)("").upsert("slashes", rows)
+          console.log('Wrote missed block events', rows.length)
+        } catch (error) {
+          console.error("Failed to write missed block events", error)
+          Sentry.captureException(error)
+        }
+      }
+    )
+  }
+}
+
+async function getLastMissedBlock(networkId, operatorAddress) {
+  // get last liveness issue
+  const {data} = await database(config)("").query(`
+    query {
+      slashes(where:{networkId:{_eq:"${networkId}"}, operatorAddress:{_eq:"${operatorAddress}"}, reason:{_eq:"missed_blocks"}}, order_by:{id:desc}, limit: 1) {
+        height
+        id
+        reason
+      }
+    }
+  `)
+
+  return data.slashes[0]
+}
```

Comment:
Another remark: I saw in DB that "slashes" is a different table. Shouldn't we have all events in a separate table of its own? Like, right now, where are we storing events in? In "notifications" and "slashes" and that is right?

Just trying to grasp why slashes get a different table here

Reply:
The idea was, that slashes are non query able. So I wanted a nondestructive way to store them. But I guess we could also just store them as notifications as the data part of the notifications also has all the data.

Reply:
or at least we should store them in both tables to stay consistent i.e. collect all events in the notifications table

Reply:
I commented that we need to find a proper strategy for storing the liveness events. If not there can easily be a couple of thousand for one validator outage.

Reply:
just an idea - log the first liveness event, start a counter, log the count, log the last liveness event.

Reply:
similar: query last event of that type (operator address and liveness) compare with current and create a new notification if triggers apply

Reply:
should I put this in here or do another PR?

Reply:
I'd put it in another one

Reply:
Oh, could you also lint this file: `api/lib/database/helpers.js`